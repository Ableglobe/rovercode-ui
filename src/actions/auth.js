// actions https://redux.js.org/basics/actions
// async actions https://redux.js.org/advanced/asyncactions

export const UPDATE_VALID_AUTH = 'UPDATE_VALID_AUTH';
export const USER_LOGOUT = 'USER_LOGOUT';

// action creators
export const updateValidAuth = isValidAuth => ({
  type: UPDATE_VALID_AUTH,
  payload: isValidAuth,
});

export const logout = () => ({
  type: USER_LOGOUT,
});
