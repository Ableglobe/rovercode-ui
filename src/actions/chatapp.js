import Axios from "axios";

export const TOGGLE_FORMS = 'TOGGLE_FORMS';
export const SET_SESSION_ID = 'SET_SESSION_ID';
export const SET_CLIENT_ID = 'SET_CLIENT_ID';
export const SET_IS_SUPPORT_PROVIDER = 'SET_IS_SUPPORT_PROVIDER';


export const toggleForms = () => ({
  type: TOGGLE_FORMS,
});

export const setClientID = id => ({
  type: SET_CLIENT_ID,
  payload: id,
});

export const setSessionID = id => ({
  type: SET_SESSION_ID,
  payload: id,
});

export const setIsSupportProvider = () => ({
  type: SET_IS_SUPPORT_PROVIDER,
});

