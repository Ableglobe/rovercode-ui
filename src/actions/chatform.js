export const CATEGORY_SELECT_CHANGE = 'CATEGORY_SELECT_CHANGE';
export const BODY_INPUT_CHANGE = 'BODY_INPUT_CHANGE';
export const EXPERIENCE_SELECT_CHANGE = 'EXPERIENCE_SELECT_CHANGE'
export const SUBJECT_INPUT_CHANGE = 'SUBJECT_INPUT_CHANGE';

// CATEGORY STATES
export const FIRST_TIME = 0;
export const PRETTY_NEW = 1;
export const FOR_A_WHILE = 2;
export const EXPERT = 3;

//EXPERIENCE STATES
export const DEBUG = "DEBUG";
export const WHAT_NEXT = "WHAT_NEXT";
export const FUN_IDEAS = "FUN_IDEAS";


export const categorySelectChange = (value) =>({
    type: CATEGORY_SELECT_CHANGE,
    payload: value
})

export const subjectInputChange = (value) =>({
    type: SUBJECT_INPUT_CHANGE,
    payload: value
})

export const bodyInputChange = (value) =>({
    type: BODY_INPUT_CHANGE,
    payload: value
})

export const experienceSelectChange = (value) =>({
    type: EXPERIENCE_SELECT_CHANGE,
    payload: value
})

