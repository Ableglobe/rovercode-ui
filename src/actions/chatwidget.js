export const TOGGLE_IN_PROGRESS_STATE = 'TOGGLE_IN_PROGRESS_STATE';

export const toggleInProgressState = () =>({
    type: TOGGLE_IN_PROGRESS_STATE,
  })