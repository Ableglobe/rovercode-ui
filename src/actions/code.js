// actions https://redux.js.org/basics/actions
// async actions https://redux.js.org/advanced/asyncactions

export const UPDATE_JSCODE = 'UPDATE_JSCODE';
export const CHANGE_EXECUTION_STATE = 'CHANGE_EXECUTION_STATE';

// Execution States
export const EXECUTION_RUN = 1;
export const EXECUTION_STEP = 2;
export const EXECUTION_STOP = 3;
export const EXECUTION_RESET = 4;

// action creators
export const updateJsCode = jsCode => ({
  type: UPDATE_JSCODE,
  payload: jsCode,
});

export const changeExecutionState = state => ({
  type: CHANGE_EXECUTION_STATE,
  payload: state,
});
