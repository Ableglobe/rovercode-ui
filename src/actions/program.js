// actions https://redux.js.org/basics/actions
// async actions https://redux.js.org/advanced/asyncactions

import axios from 'axios';

export const FETCH_PROGRAMS = 'FETCH_PROGRAMS';
export const FETCH_PROGRAMS_PENDING = `${FETCH_PROGRAMS}_PENDING`;
export const FETCH_PROGRAMS_FULFILLED = `${FETCH_PROGRAMS}_FULFILLED`;
export const FETCH_PROGRAMS_REJECTED = `${FETCH_PROGRAMS}_REJECTED`;
export const FETCH_USER_PROGRAMS = 'FETCH_USER_PROGRAMS';
export const FETCH_USER_PROGRAMS_PENDING = `${FETCH_USER_PROGRAMS}_PENDING`;
export const FETCH_USER_PROGRAMS_FULFILLED = `${FETCH_USER_PROGRAMS}_FULFILLED`;
export const FETCH_USER_PROGRAMS_REJECTED = `${FETCH_USER_PROGRAMS}_REJECTED`;
export const REMOVE_PROGRAM = 'REMOVE_PROGRAM';
export const REMOVE_PROGRAM_PENDING = `${REMOVE_PROGRAM}_PENDING`;
export const REMOVE_PROGRAM_FULFILLED = `${REMOVE_PROGRAM}_FULFILLED`;
export const REMOVE_PROGRAM_REJECTED = `${REMOVE_PROGRAM}_REJECTED`;

// action creators
export const fetchPrograms = (xhrOptions) => {
  let type = FETCH_PROGRAMS;

  if (xhrOptions && xhrOptions.params && xhrOptions.params.user) {
    type = FETCH_USER_PROGRAMS;
  }

  return ({
    type,
    meta: {
      debounce: {
        time: 500,
      },
    },
    payload: axios.get('/api/v1/block-diagrams/', xhrOptions)
      .then(({ data }) => (
        data
      )),
  });
};

export const removeProgram = (id, xhrOptions) => ({
  type: REMOVE_PROGRAM,
  payload: axios.delete(`/api/v1/block-diagrams/${id}/`, xhrOptions)
    .then(({ data }) => (
      data
    )),
});
