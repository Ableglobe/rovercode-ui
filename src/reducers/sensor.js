import {
  CHANGE_LEFT_SENSOR_STATE,
  CHANGE_RIGHT_SENSOR_STATE,
  NOT_COVERED,
} from '../actions/sensor';

export default function sensor(
  state = {
    left: NOT_COVERED,
    right: NOT_COVERED,
  },
  action,
) {
  switch (action.type) {
    case CHANGE_LEFT_SENSOR_STATE:
      return {
        ...state,
        left: action.payload,
      };
    case CHANGE_RIGHT_SENSOR_STATE:
      return {
        ...state,
        right: action.payload,
      };
    default:
      return state;
  }
}
