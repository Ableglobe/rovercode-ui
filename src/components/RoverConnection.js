import React, { Component, Fragment } from 'react';
import { Card, Label, Icon } from 'semantic-ui-react';
import { hot } from 'react-hot-loader';
import { FormattedMessage } from 'react-intl';
import PropTypes from 'prop-types';
import Websocket from 'react-websocket';
import { COVERED, NOT_COVERED } from '@/actions/sensor';

const heartbeatTimeout = 8000; // milliseconds

class RoverConnection extends Component {
  constructor(props) {
    super(props);

    this.timeout = null;
    this.state = {
      online: false,
    };
  }

  componentDidMount() {
    this.startHeartbeatTimer();
    this.sendCommand();
  }

  componentDidUpdate() {
    this.sendCommand();
  }

  componentWillUnmount() {
    clearTimeout(this.timeout);
  }

  startHeartbeatTimer = () => {
    this.timeout = setTimeout(this.setOffline, heartbeatTimeout);
  }

  setActive = () => {
    const { changeActiveRover, clientId } = this.props;

    changeActiveRover(clientId);
  }

  setOffline = () => this.setState({ online: false })

  setOnline = () => this.setState({ online: true })

  onMessage = (data) => {
    const { changeLeftSensorState, changeRightSensorState, isActive } = this.props;
    const { online } = this.state;

    const message = JSON.parse(data);

    if (message.type === 'sensor-reading') {
      if (isActive) {
        const value = message['sensor-value'];
        if (message['sensor-id'] === 'ultrasonic-left') {
          changeLeftSensorState(value ? COVERED : NOT_COVERED);
        } else if (message['sensor-id'] === 'ultrasonic-right') {
          changeRightSensorState(value ? COVERED : NOT_COVERED);
        }
      }
    } else if (message.type === 'heartbeat') {
      clearTimeout(this.timeout);
      this.startHeartbeatTimer();

      if (!online) {
        this.setOnline();
      }
    }
  }

  sendCommand = () => {
    const { commands, isActive, popCommand } = this.props;

    if (commands.length && isActive) {
      this.wsRef.sendMessage(commands[0]);
      popCommand();
    }
  }

  render() {
    const { clientId, isActive, name } = this.props;
    const { online } = this.state;

    const wsProtocol = window.location.protocol === 'https:' ? 'wss' : 'ws';
    const wsPort = window.location.protocol === 'https:' ? '443' : '8000';
    const wsUrl = `${wsProtocol}://${window.location.hostname}:${wsPort}/ws/realtime/${clientId}/`;

    return (
      <Fragment>
        <Card color={isActive ? 'blue' : null} onClick={this.setActive}>
          <Card.Content>
            <Label corner="right" style={{ borderColor: 'white' }}>
              {
                online ? (
                  <Icon name="circle" color="green" />
                ) : (
                  <Icon name="circle outline" />
                )
              }
            </Label>
            <Card.Header>
              { name }
            </Card.Header>
            <Card.Meta>
              {
                isActive ? (
                  <FormattedMessage
                    id="app.rover_connection.active"
                    description="Label indicating the rover is connected"
                    defaultMessage="Active"
                  />
                ) : (
                  <FormattedMessage
                    id="app.rover_connection.inactive"
                    description="Label indicating the rover is not connected"
                    defaultMessage="Inactive"
                  />
                )
              }
            </Card.Meta>
          </Card.Content>
        </Card>
        <Websocket
          url={wsUrl}
          onMessage={this.onMessage}
          ref={(ws) => { this.wsRef = ws; }}
        />
      </Fragment>
    );
  }
}

RoverConnection.defaultProps = {
  commands: [],
};

RoverConnection.propTypes = {
  isActive: PropTypes.bool.isRequired,
  clientId: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  changeLeftSensorState: PropTypes.func.isRequired,
  changeRightSensorState: PropTypes.func.isRequired,
  changeActiveRover: PropTypes.func.isRequired,
  popCommand: PropTypes.func.isRequired,
  commands: PropTypes.arrayOf(PropTypes.string),
};

export default hot(module)(RoverConnection);
