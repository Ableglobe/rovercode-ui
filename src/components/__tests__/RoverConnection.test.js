import React from 'react';
import { Card, Icon } from 'semantic-ui-react';
import { mount, shallow } from 'enzyme';
import Websocket from 'react-websocket';
import RoverConnection from '../RoverConnection';
import { COVERED, NOT_COVERED } from '@/actions/sensor';

let changeActiveRover;
let changeLeftSensorState;
let changeRightSensorState;
let mockWsSend;
let popCommand;

global.WebSocket = jest.fn().mockImplementation(() => ({
  send: mockWsSend,
}));

describe('The RoverList component', () => {
  beforeEach(() => {
    changeActiveRover = jest.fn();
    changeLeftSensorState = jest.fn();
    changeRightSensorState = jest.fn();
    mockWsSend = jest.fn();
    popCommand = jest.fn();
  });

  test('renders on the page with no errors', () => {
    const wrapper = shallow(
      <RoverConnection
        changeActiveRover={changeActiveRover}
        changeLeftSensorState={changeLeftSensorState}
        changeRightSensorState={changeRightSensorState}
        popCommand={popCommand}
        clientId="1234"
        name="Sparky"
        isActive
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('displays active rover', () => {
    delete global.window.location;
    global.window.location = {
      protocol: 'http:',
      hostname: 'example.com',
    };
    const wrapper = shallow(
      <RoverConnection
        changeActiveRover={changeActiveRover}
        changeLeftSensorState={changeLeftSensorState}
        changeRightSensorState={changeRightSensorState}
        popCommand={popCommand}
        clientId="1234"
        name="Sparky"
        isActive
      />,
    );

    expect(wrapper.find(Card).length).toBe(1);
    expect(wrapper.find(Card).first().prop('color')).toBe('blue');
    expect(wrapper.find(Card.Meta).first().prop('children')).toBe('Active');
    expect(wrapper.find(Card.Header).first().prop('children')).toBe('Sparky');
    expect(wrapper.find(Websocket).length).toBe(1);
    expect(wrapper.find(Websocket).prop('url')).toBe('ws://example.com:8000/ws/realtime/1234/');
  });

  test('displays inactive rover', () => {
    delete global.window.location;
    global.window.location = {
      protocol: 'https:',
      hostname: 'example.com',
    };
    const wrapper = shallow(
      <RoverConnection
        changeActiveRover={changeActiveRover}
        changeLeftSensorState={changeLeftSensorState}
        changeRightSensorState={changeRightSensorState}
        popCommand={popCommand}
        clientId="1234"
        name="Sparky"
        isActive={false}
      />,
    );

    expect(wrapper.find(Card).length).toBe(1);
    expect(wrapper.find(Card).first().prop('color')).toBeNull();
    expect(wrapper.find(Card.Meta).first().prop('children')).toBe('Inactive');
    expect(wrapper.find(Card.Header).first().prop('children')).toBe('Sparky');
    expect(wrapper.find(Websocket).length).toBe(1);
    expect(wrapper.find(Websocket).prop('url')).toBe('wss://example.com:443/ws/realtime/1234/');
  });

  test('displays online rover', () => {
    const wrapper = shallow(
      <RoverConnection
        changeActiveRover={changeActiveRover}
        changeLeftSensorState={changeLeftSensorState}
        changeRightSensorState={changeRightSensorState}
        popCommand={popCommand}
        clientId="1234"
        name="Sparky"
        isActive
      />,
    );

    wrapper.setState({
      online: true,
    });
    wrapper.update();

    expect(wrapper.find(Icon).first().prop('name')).toBe('circle');
    expect(wrapper.find(Icon).first().prop('color')).toBe('green');
  });

  test('displays offline rover', () => {
    const wrapper = shallow(
      <RoverConnection
        changeActiveRover={changeActiveRover}
        changeLeftSensorState={changeLeftSensorState}
        changeRightSensorState={changeRightSensorState}
        popCommand={popCommand}
        clientId="1234"
        name="Sparky"
        isActive
      />,
    );

    wrapper.setState({
      online: false,
    });
    wrapper.update();

    expect(wrapper.find(Icon).first().prop('name')).toBe('circle outline');
    expect(wrapper.find(Icon).first().prop('color')).toBeUndefined();
  });

  test('changes sensor state on message', () => {
    const wrapper = shallow(
      <RoverConnection
        changeActiveRover={changeActiveRover}
        changeLeftSensorState={changeLeftSensorState}
        changeRightSensorState={changeRightSensorState}
        popCommand={popCommand}
        clientId="1234"
        name="Sparky"
        isActive
      />,
    );

    wrapper.instance().onMessage(JSON.stringify({
      type: 'sensor-reading',
      'sensor-type': 'distance',
      'sensor-id': 'ultrasonic-left',
      'sensor-value': true,
      unit: 'active-high',
    }));

    wrapper.instance().onMessage(JSON.stringify({
      type: 'sensor-reading',
      'sensor-type': 'distance',
      'sensor-id': 'ultrasonic-left',
      'sensor-value': false,
      unit: 'active-high',
    }));

    wrapper.instance().onMessage(JSON.stringify({
      type: 'sensor-reading',
      'sensor-type': 'distance',
      'sensor-id': 'ultrasonic-right',
      'sensor-value': true,
      unit: 'active-high',
    }));

    wrapper.instance().onMessage(JSON.stringify({
      type: 'sensor-reading',
      'sensor-type': 'distance',
      'sensor-id': 'ultrasonic-right',
      'sensor-value': false,
      unit: 'active-high',
    }));

    expect(changeLeftSensorState).toHaveBeenCalledTimes(2);
    expect(changeLeftSensorState.mock.calls[0][0]).toBe(COVERED);
    expect(changeLeftSensorState.mock.calls[1][0]).toBe(NOT_COVERED);
    expect(changeRightSensorState).toHaveBeenCalledTimes(2);
    expect(changeRightSensorState.mock.calls[0][0]).toBe(COVERED);
    expect(changeRightSensorState.mock.calls[1][0]).toBe(NOT_COVERED);
  });

  test('ignores sensor sensor state on message with inactive', () => {
    const wrapper = shallow(
      <RoverConnection
        changeActiveRover={changeActiveRover}
        changeLeftSensorState={changeLeftSensorState}
        changeRightSensorState={changeRightSensorState}
        popCommand={popCommand}
        clientId="1234"
        name="Sparky"
        isActive={false}
      />,
    );

    wrapper.instance().onMessage(JSON.stringify({
      type: 'sensor-reading',
      'sensor-type': 'distance',
      'sensor-id': 'ultrasonic-left',
      'sensor-value': true,
      unit: 'active-high',
    }));

    wrapper.instance().onMessage(JSON.stringify({
      type: 'sensor-reading',
      'sensor-type': 'distance',
      'sensor-id': 'ultrasonic-left',
      'sensor-value': false,
      unit: 'active-high',
    }));

    wrapper.instance().onMessage(JSON.stringify({
      type: 'sensor-reading',
      'sensor-type': 'distance',
      'sensor-id': 'ultrasonic-right',
      'sensor-value': true,
      unit: 'active-high',
    }));

    wrapper.instance().onMessage(JSON.stringify({
      type: 'sensor-reading',
      'sensor-type': 'distance',
      'sensor-id': 'ultrasonic-right',
      'sensor-value': false,
      unit: 'active-high',
    }));

    expect(changeLeftSensorState).toHaveBeenCalledTimes(0);
    expect(changeRightSensorState).toHaveBeenCalledTimes(0);
  });

  test('ignores unknown sensors', () => {
    const wrapper = shallow(
      <RoverConnection
        changeActiveRover={changeActiveRover}
        changeLeftSensorState={changeLeftSensorState}
        changeRightSensorState={changeRightSensorState}
        popCommand={popCommand}
        clientId="1234"
        name="Sparky"
        isActive
      />,
    );

    wrapper.instance().onMessage(JSON.stringify({
      type: 'sensor-reading',
      'sensor-type': 'distance',
      'sensor-id': 'test-sensor',
      'sensor-value': true,
      unit: 'active-high',
    }));

    expect(changeLeftSensorState).not.toHaveBeenCalled();
    expect(changeRightSensorState).not.toHaveBeenCalled();
  });

  test('sets sensor online on message', () => {
    const wrapper = shallow(
      <RoverConnection
        changeActiveRover={changeActiveRover}
        changeLeftSensorState={changeLeftSensorState}
        changeRightSensorState={changeRightSensorState}
        popCommand={popCommand}
        clientId="1234"
        name="Sparky"
        isActive
      />,
    );

    expect(wrapper.state('online')).toBe(false);

    wrapper.instance().onMessage(JSON.stringify({
      type: 'heartbeat',
    }));

    expect(wrapper.state('online')).toBe(true);
    expect(changeLeftSensorState).not.toHaveBeenCalled();
    expect(changeRightSensorState).not.toHaveBeenCalled();

    wrapper.instance().onMessage(JSON.stringify({
      type: 'heartbeat',
    }));

    expect(wrapper.state('online')).toBe(true);
  });

  test('ignores unknown message', () => {
    const wrapper = shallow(
      <RoverConnection
        changeActiveRover={changeActiveRover}
        changeLeftSensorState={changeLeftSensorState}
        changeRightSensorState={changeRightSensorState}
        popCommand={popCommand}
        clientId="1234"
        name="Sparky"
        isActive
      />,
    );

    wrapper.instance().onMessage(JSON.stringify({
      type: 'test-type',
    }));

    expect(wrapper.state('online')).toBe(false);
    expect(changeLeftSensorState).not.toHaveBeenCalled();
    expect(changeRightSensorState).not.toHaveBeenCalled();
  });

  test('sets active on click', () => {
    const wrapper = shallow(
      <RoverConnection
        changeActiveRover={changeActiveRover}
        changeLeftSensorState={changeLeftSensorState}
        changeRightSensorState={changeRightSensorState}
        popCommand={popCommand}
        clientId="1234"
        name="Sparky"
        isActive={false}
      />,
    );

    wrapper.find(Card).simulate('click');

    expect(changeActiveRover).toHaveBeenCalledWith('1234');
  });

  test('sets offline after timeout', () => {
    jest.useFakeTimers();

    const wrapper = shallow(
      <RoverConnection
        changeActiveRover={changeActiveRover}
        changeLeftSensorState={changeLeftSensorState}
        changeRightSensorState={changeRightSensorState}
        popCommand={popCommand}
        clientId="1234"
        name="Sparky"
        isActive={false}
      />,
    );

    expect(setTimeout).toHaveBeenCalledTimes(1);

    wrapper.setState({ online: true });

    jest.runAllTimers();

    expect(wrapper.state('online')).toBe(false);
  });

  test('sends all commands when active', () => {
    const wrapper = mount(
      <RoverConnection
        changeActiveRover={changeActiveRover}
        changeLeftSensorState={changeLeftSensorState}
        changeRightSensorState={changeRightSensorState}
        popCommand={popCommand}
        clientId="1234"
        name="Sparky"
        commands={['command1', 'command2', 'command3']}
        isActive
      />,
    );

    expect(mockWsSend).toHaveBeenCalledTimes(1);
    expect(popCommand).toHaveBeenCalledTimes(1);

    wrapper.setProps({
      commands: ['command2', 'command3'],
    });

    expect(mockWsSend).toHaveBeenCalledTimes(2);
    expect(popCommand).toHaveBeenCalledTimes(2);

    wrapper.setProps({
      commands: ['command3'],
    });

    expect(mockWsSend).toHaveBeenCalledTimes(3);
    expect(popCommand).toHaveBeenCalledTimes(3);

    wrapper.setProps({
      commands: [],
    });

    expect(mockWsSend).toHaveBeenCalledTimes(3);
    expect(popCommand).toHaveBeenCalledTimes(3);
  });

  test('does not send commands when inactive', () => {
    mount(
      <RoverConnection
        changeActiveRover={changeActiveRover}
        changeLeftSensorState={changeLeftSensorState}
        changeRightSensorState={changeRightSensorState}
        popCommand={popCommand}
        clientId="1234"
        name="Sparky"
        commands={['command1', 'command2', 'command3']}
        isActive={false}
      />,
    );

    expect(mockWsSend).toHaveBeenCalledTimes(0);
    expect(popCommand).toHaveBeenCalledTimes(0);
  });
});
