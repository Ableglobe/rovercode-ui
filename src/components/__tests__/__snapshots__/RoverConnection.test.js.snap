// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The RoverConnection component renders on the page with no errors 1`] = `
<Fragment>
  <Card
    color="blue"
    onClick={[Function]}
  >
    <CardContent>
      <Label
        corner="right"
        style={
          Object {
            "borderColor": "white",
          }
        }
      >
        <Icon
          as="i"
          name="circle outline"
        />
      </Label>
      <CardHeader>
        Sparky
      </CardHeader>
      <CardMeta>
        <FormattedMessage
          defaultMessage="Active"
          id="app.rover_connection.active"
          values={Object {}}
        />
      </CardMeta>
    </CardContent>
  </Card>
  <Websocket
    debug={false}
    onMessage={[Function]}
    reconnect={true}
    url="ws://localhost:8000/ws/realtime/1234/"
  />
</Fragment>
`;
